### ISSUE TITLE
AVIRIS L2 Scenes activation module

### ISSUE THREAD

**Issue by pomadchin** [azavea/nasa-hyperspectral#55]
- Download AVIRIS L2 scenes from the FTP and Cogify them
    - Query Franklin to retrieve AVIRIS scene(s)
    - Convert it into pixel interleave COGs via GDAL
    - Upload scene(s) to S3 and add Items into a STAC Catalog (in a separate AVIRIS Activated catalog)

### 12/7/2020 Update

Core functionality is done. Remaining TODOs after a brief discussion with @pomadchin:
- [x] Switch COG generation to DEFLATE compression
- [x] Switch activator to take AVIRIS STAC ID rather than AVIRIS Name
- [x] Update AVIRIS STAC Catalog generation script to generate a flat catalog with no nested collections
- [x] Delete and re-upload AVIRIS STAC Catalog to Franklin instance
- [x] Open new issue for task to skip activator processing if already processed
- [ ] ~~Switch to logger~~
- [x] Raise exceptions on error, exit zero on success
- [x] Verify gdalinfo can remote read .tiff.aux.xml

**Comment by CloudNiner** [comment]
Ok. I pulled one of the gzipped archives, and it looks like there's a number of files in each archive:

```
   *eph                  position data in a WGS-84/NAD83 UTM x,y,z coordinate system,
   *lonlat_eph           position in WGS-84 longitude, latitude and elevation,
   *obs                  parameters relating to the geometry of the observation and
                         illumination conditions in raw spatial format,
   *obs.hdr              OBS image header file
   *obs_ort              parameters relating to the geometry of the observation and
                         illumination conditions rendered using the *_ort_glt lookup
                         table,
   *obs_ort.hdr          OBS_ORT image header file,
   *ort.plog             generic data processing info
   *ortho.readme         this file,
   *ort_glt              geometric look-up table,
   *ort_glt.hdr          GLT image header file,
   *ort_igm              input geometry file,
   *ort_igm.hdr          IGM image header file,
   *ort_img              orthocorrected, scaled radiance image file,
   *ort_img.hdr          ORT_IMG image header file.
``` 

The largest file is `*ort_img` which is the HDR file we were originally planning to COGify and copy, so this extra effort should keep bandwith, storage and running time costs approximately on the same order of magnitude as originally planned.

I think, within the scope of this task, I can COGify each of the `.hdr` files and then copy all COG files + any remaining files as is to S3. _With that said_, if we don't have a known use case for any of the other files we should just skip them for now to keep complexity low. We can always update the activator later and invalidate the cache to regenerate it with the extra files.

**Comment by CloudNiner** [comment]
Actually, now that I'm thinking about it -- @jpolchlo and @pomadchin do we need the AVIRIS refl L2 imagery first? The discussion above is predicated on pulling and COGifying the regular AVIRIS imagery.

**Comment by pomadchin** [comment]
@CloudNiner my vote would be for both L1 and L2, since both of these files can be useful for us, and it may allow @jamesmcclain to access L1 data efficiently as well. 

Mb it makes sense to specify the type of imagery to process as a cli param if there is a worry about network and processing times? i.e. L1 / L2 / both

I thought that it doesnt really matter what files to convert via GDAL, or there is some critical difference between L1 and L2 COGifications?

**Comment by CloudNiner** [comment]
The L2 refl archive is a pair of hdr files that would be cogified in the same way as far as I can tell. I haven't written the activator yet so it's tough to say if it's better to control this via a CLI argument or just have two separate activators. I'd currently lean towards two separate activators to keep each activator as much of a single unit of work as possible that adds the files it generates to a single STAC collection. 

There's definitely scope creep happening here, so for this issue we should pick L1 or L2, whichever we anticipate needing first. The other type can be a follow up issue.

**Comment by pomadchin** [comment]
@CloudNiner :+1: got your plan, makes sense to me. 

As far as I understand L2 is what we need for @jpolchlo => let's focus on that; summoning @jpolchlo for confirmation.

**Comment by jpolchlo** [comment]
Having L2 access will be very useful.  Sounds like @jamesmcclain can probably either wait or download imagery directly for his own purposes?

**Comment by CloudNiner** [comment]
L2 it is!

**Comment by pomadchin** [comment]
Important metadata that should be collected:
- Collect bands frequences (centers of wavelengths)

There are lots of sensors that could have changed. 
A note: this data can be collected only from the L1 scene.

Someone would need to review what metadata is neccesary and where it is stored. (it would be a separate task, upd the link later here)
TLDR; that was a very good point about trying to preserve all possible metadata.
